apps/antalmanac/src/components/RightPane/CoursePane/CoursePaneButtonRow.tsx

```diff
@@ -65,20 +66,24 @@ export function ColumnToggleButton() {
         store.selectedColumns,
         store.setSelectedColumns,
     ]);
-    const [open, setOpen] = useState(false);
+    const [anchorEl, setAnchorEl] = useState<HTMLElement>();
+    const open = useMemo(() => Boolean(anchorEl), [anchorEl]);
 
-    const handleChange = useCallback((e: SelectChangeEvent<SectionTableColumn[]>) => {
-        if (typeof e.target.value !== 'string') {
-            setSelectedColumns(e.target.value);
-        }
-    }, []);
+    const handleChange = useCallback(
+        (e: SelectChangeEvent<SectionTableColumn[]>) => {
+            if (typeof e.target.value !== 'string') {
+                setSelectedColumns(e.target.value);
+            }
+        },
+        [setSelectedColumns]
+    );
 
-    const handleOpen = useCallback(() => {
-        setOpen(true);
-    }, [setOpen]);
+    const handleClick = useCallback((event: React.MouseEvent<HTMLButtonElement>) => {
```

Comment:
You can generalize this type so it's more succinct, since we don't really need accuracy in this function. A similiar, maybe synonymous type might be `React.SyntheticEvent`

```suggestion
    const handleClick = useCallback((event: React.MouseEvent) => {
```

Reply:
Argument of type 'EventTarget & Element' is not assignable to parameter of type 'SetStateAction<HTMLElement | undefined>'.
  Type 'EventTarget & Element' is missing the following properties from type 'HTMLElement': accessKey, accessKeyLabel, autocapitalize, dir, and 122 more.ts(2345)

```typescript
    const handleClick = useCallback((event: React.MouseEvent) => {
        setAnchorEl(event.currentTarget);
    }, []);
```

I think it's probably okay to specify HTMLButtonElement here >.<